tank/examples/opt/opt_perf_comparison.py

```diff
@@ -96,39 +119,53 @@ def create_vmfb_module(
     # compilation time can be correctly measured only when MLIR has already been
     # loaded.
     assert not recompile_shark or has_mlir
-    if has_mlir:
-        with open(mlir_path, "r") as f:
-            model_mlir = f.read()
-        print(f"Loaded .mlir from {mlir_path}")
-    else:
-        (model_mlir, func_name) = import_with_fx(
-            model=opt_model,
-            inputs=inputs,
-            is_f16=False,
-            model_name=opt_fs_name,
-            return_str=True,
+    if not has_mlir:
+        import_mlir_module(
+            model_name,
+            tokenizer,
+            device,
+            max_seq_len,
         )
-        with open(mlir_path, "w") as f:
-            f.write(model_mlir)
-        print(f"Saved mlir at {mlir_path}")
-
-    shark_module = SharkInference(
-        model_mlir,
-        device=device,
-        mlir_dialect="tm_tensor",
-        is_benchmark=False,
-    )
+    # with open(mlir_path, "r") as f:
+    #    model_mlir = f.read()
+    # print(f"Loaded .mlir from {mlir_path}")
+    # breakpoint()
+    # shark_module = SharkInference(
+    #    model_mlir,
+    #    device=device,
+    #    mlir_dialect="tm_tensor",
+    #    is_benchmark=False,
+    #    rt_flags=[],
+    # )
 
     vmfb_name = (
         f"{opt_fs_name}_causallm_{max_seq_len}_torch_{DEVICE}_tiled_ukernels"
     )
-    shark_module.save_module(module_name=vmfb_name)
+    # shark_module.save_module(module_name=vmfb_name)
     vmfb_path = vmfb_name + ".vmfb"
+    ireec.compile_file(
```

Comment:
we dont have an abstraction for this?

Reply:
This was proof of concept for the new path through shark inference (compile_str --> compile_file).
Updated it, thanks!